arangod/VocBase/Methods/Collections.cpp

```diff
@@ -130,7 +130,7 @@ void Collections::enumerate(TRI_vocbase_t* vocbase,
 /*static*/ arangodb::Result methods::Collections::lookup( // find collection
     TRI_vocbase_t const& vocbase, // vocbase to search
     std::string const& name, // collection name
-    FuncCallback const& func // invoke on found collection
+    FuncCallback func // invoke on found collection
```

Comment:
it's usually good practice to use 'const&' with lambdas to avoid making a copy of a lambda

Reply:
I tend to disagree. While it is true that this avoids copies in many cases this imposes other limitations, e.g., the `operator()` has to be const. In general I tend to pass functors by value and use `std::move` to avoid copying.
(FWIW: the STL also uses pass-by-value for functors since this is the most universal approach)

Reply:
The typedef for FuncCallback already contains const&

Reply:
oh, ok
I don't usually expect to see a reference (&) as part of a typedef declaration so I didn't look, sorry

Reply:
```
@mpoeter 
> In general I tend to pass functors by value and use std::move to avoid copying.
sometimes/often the lambdas can be made static and thus cannot be moved
```

Reply:
A static lambda? The big advantage of lambdas is that they can capture variables, but for static lambdas this probably doesn't make sense. In such cases it might make more sense to use a normal (or possibly static) function instead. Anyway, there is no strict rule here..! In my experience pass-by-value is the most universal approach for functors, and if the callers are careful to use `std::move` where appropriate, it comes at virtually no cost.
But const-ref functors also work fine in many cases, so if you are not concerned by its limitations just use it. No judgement here. :slightly_smiling_face: